src/Fable.AST/Fable.fs

```diff
@@ -482,6 +482,29 @@ type ArrayKind =
     | MutableArray
     | ImmutableArray
 
+[<RequireQualifiedAccess>]
+type NumberValue =
+    | Int8 of sbyte
+    | UInt8 of byte
+    | Int16 of int16
+    | UInt16 of System.UInt16
+    | Int32 of System.Int32
+    | UInt32 of System.UInt32
+    | Int64 of System.Int64
+    | UInt64 of System.UInt64
+    | Int128 of System.Int64 // System.Int128
+    | UInt128 of System.UInt64 // System.UInt128
+    | BigInt of bigint
+    | NativeInt of nativeint
+    | UNativeInt of unativeint
+    | Float16 of System.Single // System.Half
+    | Float32 of System.Single
+    | Float64 of System.Double
+    | Decimal of System.Decimal
+    // this will be coerced to type later / depends on lang
+    // TODO: review KindOfChar, it reeks of poor modelling
+    | KindOfChar of NumberKind * System.Char
```

Comment:
I agree that doesn't make sense, when there is `CharConstant`.
Conversions from number to char can be expressed as `TypeCast`.